### ISSUE TITLE
Pressing F1 at gnome desktop or at libreoffice or  at nautilus window... etc, not working. 

### ISSUE THREAD

**Issue by franzhuang** [lanoxx/tilda#474]
I have Fedora 34 , Gnome 40 running on my computer.
Tilda drop down function works well when I use web browser, but it doesn't work when I am on desktop or libreoffice or nautilus window...etc.
Is there way to fix this?

Thank you.

**Comment by tricertc** [comment]
I was having the same issue, #407 resolved it for me.  There's a lot of helpful information in that conversation, but at a high level I had to

* clone the repo
* build/install from the master branch
* map a global hot key in GNOME settings 
* launch tilda with `tilda --dbus`

**Comment by franzhuang** [comment]
Thank you.
I switched to gnome Xorg from gnome Wayland, and the problem resolved.
I should read README first.

**Comment by akukalev** [comment]
Hello, is there any hope to fix this bug? All drop down terminals have this issue since Ubuntu 22.04 was released. This super annoying on every day basis. Thank you.

**Comment by eugenioperea** [comment]
I've been having this problem for a long time, but I got used to it. I'd just alt-tab to an app where it does work, but now Firefox and Thunderbird also grab F1 before Tilda can. Is there a workaround I can use?

**Comment by KingJames0000** [comment]
I too have this issue. I'm using wayland, that may have something to do with it, but none of the fixes worked. I tried @tricertc's suggestions, except for the custom build part, but it didn't work. Is it neccesary to custom build it?

**Comment by lanoxx** [comment]
Only if your Linux distribution does not yet have tilda 2.0.0. If you are using Debian Testing, Unstable or the upcoming Ununtu 24.04, then you should have tilda 2.0.0, older distributions still have tilda 1.x. I am not sure about other distributions like fedora.

**Comment by KingJames0000** [comment]
I may be an outlier, but I'm running Debian Testing alongside Manjaro with the unstable kernel, both with versions of Tilda equal to or greater than 2.0, and I can't get it to work. 

After doing some reading, I am guessing that this may be a direct result of the Wayland protocol. Most apps haven't been ported from X, as I understand,  and Wayland has very strict key stroke rules. I am also crippled by having a extremely limited selection of on screen keyboards for the same reason.

**Comment by lanoxx** [comment]
Does `tilda —toggle` pull the terminal? If that works, then next check if you have correctly registered a custom hotkey in the desktop environment. I have tested with Gnome Shell, but this should also work on other desktops. Normally i would expect, that the wayland session intercepts all keystrokes through the desktop shell and thus should be able to identify the configured hotkey. The question is if its the hotkey itself that is not working, or if its the dbus toggle action that has a problem. 

Also you need to start tilda with the —dbus option, and keep in mind that on when tilda is started with —dbus, then the hotkey option for the pull toggle in they preferences has no effect. This should all be explained in the new man page. If anything is unclear there, then please let me know and we can extend the documentation. 

If this is a bug with your specific environment then we need to narrow the cause down a bit more to understand where exactly this is failing.

**Comment by KingJames0000** [comment]
Well, it almost worked 😂

If I type  `tilda -T` , `tilda --toggle-window` , or append specifically for instance 0 by `tilda -T 0`  into a command prompt, it works. However, when I add any of the above to a Plasma shortcut in settings, I don't get a response.

You may be right, it may be a KDE Plasma problem 🤷‍♂️

**Comment by KingJames0000** [comment]
Correction!! I spoke too soon! After rebooting, it works as expected! Thanks ever so much!

**Comment by akukalev** [comment]
> You may be right, it may be a KDE Plasma problem 🤷‍♂️

I do not think its a KDE problem because I'm having it on Ubuntu gnome and --toogle-window  is not even a recognized option for me. Well, I had to switch to other terminal apps because tilda does not properly work for years by now.

**Comment by KingJames0000** [comment]
I know for me it only works when tilda is started with dbus (I believe this is a separate package). Was tilda already running with dbus when you tried `tilda --toggle-window`?

**Comment by ram-on** [comment]
For future readers, I've fixed this issue as follows (for GNOME):

1. Use GNOME Tweaks > Startup Applications > Add Tilda D-BUS.
2. Settings > Keyboard > View and Customize Shortcuts > Custom Shortcuts > Add a new shortcut with the following command:  tilda -T
3. Log out + login or just start tilda d-bus app.

@lanoxx  While this hack works, this is still an issue as users expect tilda to handle this by setting the "Pull down terminal" option -- such option is currently ignored for GNOME, which is therefore a bug.

**Comment by akukalev** [comment]
> For future readers, I've fixed this issue as follows (for GNOME):

This is fantastic!! Thanks so much, it works indeed and I finally got back my favorite terminal. It's a pity that developers did not care and did not provide any help, when obviously all GNOME users were affected and could not properly use tilda anymore for several years (!).

**Comment by eugenioperea** [comment]
I disagree that "developers did not care and did not provide any help". They certainly did, and proof of that is that they created a d-bus interface, as explained here: [Supported Platforms](https://github.com/lanoxx/tilda#supported-platforms). That's exactly how I got tilda working again in Ubuntu 24.04.
